Code/Tools/ProjectManager/Source/ProjectExportController.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <ProjectInfo.h>
+
+#include <QThread>
+#endif
+
+QT_FORWARD_DECLARE_CLASS(QProcess)
+
+namespace O3DE::ProjectManager
+{
+    QT_FORWARD_DECLARE_CLASS(ProjectButton)
+    QT_FORWARD_DECLARE_CLASS(ProjectExportWorker)
+
+    class ProjectExportController : public QObject
+    {
+        Q_OBJECT
+
+    public:
+        explicit ProjectExportController(const ProjectInfo& projectInfo, ProjectButton* projectButton, QWidget* parent = nullptr);
+        ~ProjectExportController();
+
+        void SetProjectButton(ProjectButton* projectButton);
+        const ProjectInfo& GetProjectInfo() const;
+
+        inline constexpr static int s_maxDisplayedBuiltOutputChars = 25;
+        inline static const char * LauncherExportFailedMessage = "Launcher failed to export.";
```

Comment:
nit: no real reason to declare these as `inline`

Reply:
the second one needs to be inline as it's a raw char string, which otherwise causes a compiler error. I tried making it a Qstring, but for some reason the QString interpolation API only works with raw strings, which I find odd.